src/main/java/org/spongepowered/mod/mixin/core/fml/common/MixinLoadController.java

```diff
@@ -65,4 +69,21 @@ public ModContainer getActiveModContainer() {
     public void setActiveModContainer(ModContainer container) {
         this.activeContainer = container;
     }
+
+    /**
+     * @author dualspiral - 4th June 2017
+     *
+     * Sponge can post events from threads that are not on the server thread,
+     * and this can cause confusion for the majority of events that are
+     * firing on the server thread. In these cases, we just get the
+     * mod from the stack.
+     *
+     * @param cir
+     */
+    @Inject(method = "activeContainer", at = @At("HEAD"), cancellable = true)
+    private void onActiveContainerHead(CallbackInfoReturnable<ModContainer> cir) {
+        if (!SpongeImpl.getServer().isCallingFromMinecraftThread()) {
```

Comment:
It is. If we're not on the minecraft server thread, _always_ find the container from the stack.